ports/xmp-toolkit-sdk/portfile.cmake

```
@@ -0,0 +1,99 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO adobe/XMP-Toolkit-SDK
+    REF v2023.12
+    SHA512 1ab987cdf50fdd0d28e2d0f97dba3cf30fe23dd1ff700d39bdf1fda7c3ae6ce1aa71806dbd77cec84ffcd7672a0a4545007c9dbcac99c83d17b4c1aa7d6b31fb
+    PATCHES
+        prepare-for-getting-zlib-and-expat-from-vcpkg.patch
+)
+
+set(pdf_handler_mini_pdfl_dir "windows")
+set(pdf_handler_resource_dir "win")
+
+set(plugin_sdk_directory "${CURRENT_PACKAGES_DIR}/src/${PORT}/plugin-sdk")
+file(MAKE_DIRECTORY "${plugin_sdk_directory}/XMPFilesPlugins/PDF_Handler/resource")
+file(COPY "${SOURCE_PATH}/XMPFilesPlugins/api" DESTINATION "${plugin_sdk_directory}/XMPFilesPlugins")
+file(COPY "${SOURCE_PATH}/XMPFilesPlugins/PluginTemplate" DESTINATION "${plugin_sdk_directory}/XMPFilesPlugins")
+file(COPY "${SOURCE_PATH}/XMPFilesPlugins/PDF_Handler/resource/${pdf_handler_resource_dir}" DESTINATION "${plugin_sdk_directory}/XMPFilesPlugins/PDF_Handler/resource")
+file(COPY "${SOURCE_PATH}/XMPFilesPlugins/PDF_Handler/${pdf_handler_mini_pdfl_dir}" DESTINATION "${plugin_sdk_directory}/XMPFilesPlugins/PDF_Handler")
+file(COPY "${SOURCE_PATH}/source" DESTINATION "${plugin_sdk_directory}")
+file(COPY "${SOURCE_PATH}/public" DESTINATION "${plugin_sdk_directory}")

```

```suggestion
set(plugin_sdk_directory "${CURRENT_PACKAGES_DIR}/src/${PORT}/plugin-sdk")
file(MAKE_DIRECTORY "${plugin_sdk_directory}/XMPFilesPlugins/PDF_Handler/resource")
file(COPY "${SOURCE_PATH}/XMPFilesPlugins/api" DESTINATION "${plugin_sdk_directory}/XMPFilesPlugins")
file(COPY "${SOURCE_PATH}/XMPFilesPlugins/PluginTemplate" DESTINATION "${plugin_sdk_directory}/XMPFilesPlugins")
file(COPY "${SOURCE_PATH}/XMPFilesPlugins/PDF_Handler/resource/${pdf_handler_resource_dir}" DESTINATION "${plugin_sdk_directory}/XMPFilesPlugins/PDF_Handler/resource")
file(COPY "${SOURCE_PATH}/XMPFilesPlugins/PDF_Handler/${pdf_handler_mini_pdfl_dir}" DESTINATION "${plugin_sdk_directory}/XMPFilesPlugins/PDF_Handler")
file(COPY "${SOURCE_PATH}/source" DESTINATION "${plugin_sdk_directory}")
file(COPY "${SOURCE_PATH}/public" DESTINATION "${plugin_sdk_directory}")

```
Why is this part of the source code copied to the installation directory package? Is there any special use for this?

---

The XMP Toolkit ships with handlers for different file formats. In addition, custom handlers ("plug-ins") can be written using the XMPFiles Plug-in SDK. The `XMPFilesPlugins` folder provides the support code and resources you need to build such plug-ins.

---

Added a comment in `portfile.cmake` for clarification.